src/main/java/org/jboss/forge/addon/swarm/ui/WildflySwarmAddFractionCommand.java

```diff
@@ -0,0 +1,85 @@
+package org.jboss.forge.addon.swarm.ui;
+
+import javax.inject.Inject;
+
+import org.jboss.forge.addon.facets.FacetFactory;
+import org.jboss.forge.addon.facets.FacetNotFoundException;
+import org.jboss.forge.addon.projects.Project;
+import org.jboss.forge.addon.projects.ProjectFactory;
+import org.jboss.forge.addon.projects.ui.AbstractProjectCommand;
+import org.jboss.forge.addon.swarm.facet.WildflySwarmFacet;
+import org.jboss.forge.addon.ui.context.UIBuilder;
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.input.InputComponentFactory;
+import org.jboss.forge.addon.ui.input.UISelectMany;
+import org.jboss.forge.addon.ui.metadata.UICommandMetadata;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+
+public class WildflySwarmAddFractionCommand extends AbstractProjectCommand {
+
+   @Inject
+   private FacetFactory facetFactory;
+
+   @Inject
+   private ProjectFactory projectFactory;
+
+   @Inject
+   private InputComponentFactory factory;
+
+   private UISelectMany<String> fractionElements;
+
+   @Override
+   public UICommandMetadata getMetadata(UIContext context)
+   {
+      return Metadata.forCommand(getClass()).name("Wildfly-Swarm: Add Fraction")
+            .category(Categories.create("Wildfly-Swarm")).description("Add one or more fractions. Installed fractions have been filtered out.");
+   }
+
+   @Override
+   public void initializeUI(UIBuilder builder) throws Exception
+   {
+      fractionElements = factory.createSelectMany("fractions", String.class).setLabel("Fraction List")
```

Comment:
Very good! It is also possible to get an `InputComponentFactory` instance by calling `UIBuilder.getInputComponentFactory()`